ui/admin/app/components/chat/ChatContext.tsx

```diff
@@ -232,3 +230,31 @@ function useMessageSource(threadId?: Nullish<string>) {
 
     return { messages, isRunning };
 }
+
+const findIndexLastPendingToolCall = (messages: Message[]) => {
+    for (let i = messages.length - 1; i >= 0; i--) {
+        const message = messages[i];
+        if (message.tools && !message.tools[0].output) {
```

Comment:
nit: is `.output` the raw output of the tool being called? if so, is it possible for a tool that has finished running to have no output here?

Reply:
Will do some verification to make sure! But it looked like `output` was always a summarized response what it was able to do with the tool and always provided at the finishing of the tool. (Ex. `"Created dataset with ID gds://4de0a with 12 issues\n{\"id\":\"gds://4de0a\",\"name\..."` or `"Sent message successfully"`)